src/Game.cpp

```diff
@@ -348,12 +354,35 @@ void Game::Battle() {
         _Player->DisplayHUD(_Enemy);
         _Enemy->DisplayHUD();
         int damagePlayer = _Player->Action();
+        int extradamagePlayer = _Player->ReturnExtraMolotovDamage();
         // Player's turn to attack Enemy or choose other action.
-        if (damagePlayer != SKIP_TURN) {
+        if (damagePlayer != SKIP_TURN && damagePlayer != 100) {
             _Enemy->TakeDamage(damagePlayer);
             // Pauses console and ignores user input for SLEEP_MS milliseconds.
             Sleep(SLEEP_MS);
         }
+        // Player turn is over, enemy's status effect damage is calculated
+        if (0 < extradamageturns && extradamageturns <= 2) {
+            extradamageturns--;
+            _Enemy->TakeDamage(extradamagePlayer);
+            Console::GetInstance().SetColour(Console::EColour::Red);
+            cout << extradamagePlayer;
+            Console::GetInstance().SetColour(Console::EColour::Default);
+            cout << " damage points from burn!" << endl;
+        }
+        // Check if enemy has had molotov thrown at him before or now
+        // Deal damage if molotov has been thrown and set extradamageturns=2
+        if (damagePlayer == 100) {
+            damagePlayer = Common::RandomInt(30, 40);
+            _Enemy->TakeDamage(damagePlayer);
+            Console::GetInstance().SetColour(Console::EColour::Red);
+            cout << damagePlayer;
+            Console::GetInstance().SetColour(Console::EColour::Default);
+            cout << " damage points!" << endl;
+            extradamageturns = 2;
+            cout << extradamageturns;
+            cout << " turns of extra burn damage" << endl;
+        }
```

Comment:
Is there any way to take this logic out of the main game loop? Ideally it shouldn't be coupled this way, and instead the scope of the extra damage logic should be limited to the player and the enemy class.

Reply:
I'm currently working on this right now.